packages/convex-helpers/server/cors.test.http.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * This file is used to define the HTTP routes for the cors.test.ts file.
+ * It does not contain any tests, but the .test path both excludes it from the
+ * generated API spec and indicates its intent.
+ */
+import { httpRouter, httpActionGeneric } from "convex/server";
+import { corsRouter } from "./cors.js";
+
+const everythingHandler = httpActionGeneric(async () => {
+  return new Response(JSON.stringify([{ fact: "Hello, world!" }]), {
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+});
+
+const http = httpRouter();
+const cors = corsRouter(http, {
+  allowedOrigins: ["*"],
+});
+
+/**
+ * Exact routes will match /fact exactly
+ */
+cors.route({
+  path: "/fact",
+  method: "GET",
+  handler: everythingHandler,
+});
+
+cors.route({
+  path: "/fact",
+  method: "POST",
+  handler: everythingHandler,
+});
+
+cors.route({
+  path: "/fact",
+  method: "PATCH",
+  handler: everythingHandler,
+});
+
+cors.route({
+  path: "/fact",
+  method: "DELETE",
+  handler: everythingHandler,
+});
+
+/**
+ * Non-CORS routes
+ */
+http.route({
```

Comment:
Can you do things like route to `/foo` with method GET with CORS and `/foo` with method POST without CORS? From a quick scan, looks like yes. I guess the OPTIONS will only reflect the ones with the CORS router, which is fine, but also probably not a super common case

Reply:
Actually that's a bit of a rough edge: I believe if you route w/o CORS, then with CORS, the CORS router will include the non-cors method in the accepted methods

Reply:
I'll file an issue to track it